### ISSUE TITLE
[BUG] write_raw_bids conversion to bv breaks event timing

### ISSUE THREAD

**Issue by alexrockhill** [mne-tools/mne-bids#293]
See

```
import os.path as op

import mne
from mne.utils import _TempDir
from mne.datasets import testing
from mne_bids import write_raw_bids, make_bids_basename


base_path = op.join(op.dirname(mne.__file__), 'io')
subject_id = '01'
subject_id2 = '02'
session_id = '01'
run = '01'
acq = '01'
run2 = '02'
task = 'testing'

bids_basename = make_bids_basename(
    subject=subject_id, session=session_id, run=run, acquisition=acq,
    task=task)

output_path = _TempDir()
data_path = testing.data_path()
raw_fname = op.join(data_path, 'MEG', 'sample',
                    'sample_audvis_trunc_raw.fif')

event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
            'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
events_fname = op.join(data_path, 'MEG', 'sample',
                       'sample_audvis_trunc_raw-eve.fif')

raw = mne.io.read_raw_fif(raw_fname)
raw.load_data()
events = mne.find_events(raw)
raw2 = raw.pick_types(meg=False, eeg=True, stim=True, eog=True, ecg=True)
raw2.save(op.join(output_path, 'test-raw.fif'), overwrite=True)
raw2 = mne.io.Raw(op.join(output_path, 'test-raw.fif'), preload=False)

write_raw_bids(raw2, bids_basename, output_path,
               verbose=True, overwrite=True)

bids_dir = op.join(output_path, 'sub-%s' % subject_id,
                   'ses-%s' % session_id, 'eeg')
raw2 = mne.io.read_raw_brainvision(op.join(bids_dir,
                                           bids_basename + '_eeg.vhdr'))

events2 = mne.find_events(raw2)

print(events, events2)
```

**Comment by alexrockhill** [comment]
Also the testing data still has an event channel instead of annotations in it, which should probably be fixed.

**Comment by jasmainak** [comment]
Why do you use `read_raw_brainvision` instead of `read_raw_bids` ?

**Comment by alexrockhill** [comment]
That's how it was in the test already but I should use that.

**Comment by alexrockhill** [comment]
Also converting to BV using pybv causes pretty non-insignificant changes in times
```
9

In [11]: raw.times[111]                                                                                   
Out[11]: 0.36962114434301757
```

Where raw2 is after conversion.

**Comment by jasmainak** [comment]
see how we read read BIDS data in the example. You should use the annotations for extracting the events. If the problem still persists, it probably needs a fix.

**Comment by alexrockhill** [comment]
Try

```
import os.path as op

import mne
from mne.utils import _TempDir
from mne.datasets import testing
from mne_bids import write_raw_bids, make_bids_basename


base_path = op.join(op.dirname(mne.__file__), 'io')
subject_id = '01'
subject_id2 = '02'
session_id = '01'
run = '01'
acq = '01'
run2 = '02'
task = 'testing'

bids_basename = make_bids_basename(
    subject=subject_id, session=session_id, run=run, acquisition=acq,
    task=task)

output_path = _TempDir()
data_path = testing.data_path()
raw_fname = op.join(data_path, 'MEG', 'sample',
                    'sample_audvis_trunc_raw.fif')

event_id = {'Auditory/Left': 1, 'Auditory/Right': 2, 'Visual/Left': 3,
            'Visual/Right': 4, 'Smiley': 5, 'Button': 32}
events_fname = op.join(data_path, 'MEG', 'sample',
                       'sample_audvis_trunc_raw-eve.fif')

raw = mne.io.read_raw_fif(raw_fname)
raw.load_data()
events, event_id = mne.events_from_annotations(raw)
print(events)
```

Output
```
[]
```

**Comment by jasmainak** [comment]
You are still not using the function I mentioned

**Comment by alexrockhill** [comment]
> You are still not using the function I mentioned

Is there a function other than events_from_annotations?

**Comment by jasmainak** [comment]
The BIDS reader will automatically populate the annotations in the `raw` object.

**Comment by alexrockhill** [comment]
The most recent comment is a separate is pre-BIDS, this is just basic mne testing data.

**Comment by alexrockhill** [comment]
There's two issues, the testing data is out of date and has the events stored as an event channel which is the second block of example code and pybv doesn't save out events correctly which is the first block.

**Comment by jasmainak** [comment]
okay so neither of them are `mne-bids` issues?

**Comment by jasmainak** [comment]
I'm still curious why you didn't try what I suggested -- use `read_raw_bids` and `events_from_annotations`. Does it give the right result or not?

**Comment by alexrockhill** [comment]
The first is mne-bids, the second is mne-python. I can open a separate PR there, but it's relevant to the current issue. I tried what you suggested, it doesn't work.

**Comment by jasmainak** [comment]
okay so I'd suggest:

1. Fix `read_raw_bids` so that it works. It's the recommended way to get events from BIDS folders.
2. Raise an issue on `pybids` page about the events

Not sure what you mean by testing data having annotations instead of event channel.

**Comment by alexrockhill** [comment]
Re: Not sure what you mean by testing data having annotations instead of event channel.

Run the block of code above, you don't get events. Why is that? If you run with find_events you do.

**Comment by jasmainak** [comment]
It's because the sample data was recorded 10 years ago and you don't store annotations in `fif`. AFAIK, annotations are only for BIDS data or annotated data or non-fif data like EDF.

**Comment by alexrockhill** [comment]
Ok, I thought all events were stored in annotations going forward.

**Comment by jasmainak** [comment]
I don't think so

**Comment by alexrockhill** [comment]
After lots of trial and error, I figured out that the stim channel in raw was overwriting the events that were passed. Should probably be a pybv issue too. Writing a fix now.

**Comment by jasmainak** [comment]
Thanks!

**Comment by alexrockhill** [comment]
Ok, sorry not fixed, something is messed up with find_events and events_from_annotations I think. Maybe someone who wrote one of those could help out.

https://github.com/mne-tools/mne-bids/issues/164 this is what I was referencing about find_events being depreciated. As far as I can tell the sample data doesn't have an analog channel (pick_types(raw.info, stim=True) returns an empty list)

**Comment by jasmainak** [comment]
closed by #294
